ChangeLog

```diff
@@ -4,6 +4,110 @@ brevity.  Much more detail can be found in the git revision history:
 
     https://github.com/jemalloc/jemalloc
 
+* 5.2.0 (TBD, 2019)
+
+  This release includes a few notable improvements, which are summarized below:
+  1) improved fast-path performance from the optimizations by @djwatson; 2)
+  reduced virtual memory fragmentation and metadata usage; and 3) bug fixes on
+  setting the number of background threads.  In addition, peak / spike memory
+  usage is improved with certain allocation patterns.  As usual, the release and
+  prior dev versions have gone through large-scale production testing.
+
+  New features:
+  - Implement oversize_threshold, which uses a dedicated arena for allocations
+    crossing the specified threshold to reduce fragmentation.  (@interwq)
+  - Add extents usage information to stats.  (@tyleretzel)
+  - Log time information for sampled allocations.  (@tyleretzel)
+  - Support 0 size in sdallocx.  (@djwatson)
```

Comment:
Is this a documented guarantee (i.e. an ABI strengthening)? I thought we were doing it to support our own operator delete overloads, but not otherwise promising anything.

Reply:
The manual does say `The minimum valid input size is the original requested size of the allocation`, to me it sounds a requirement to support 0 (as it can be the original size of malloc). No strong opinion though. Do you think we should not mention it here?

Reply:
That sounds convincing to me.

Reply:
I was alarmed when I read this particular item as well.  The `*allocx` API has always been intended as providing precisely the semantics needed by a well written application.  IMO the memory allocator shouldn't be universally burdened with sanity-checking bogus inputs.  If our sized deallocation support for C++ needs this guarantee, fine, but I would really rather not add this to the `sdallocx` spec.